### ISSUE TITLE
[v2.3.0][v2.2.7] Certificates continue to be marked as expired after Certification Rotation

### ISSUE THREAD

**Issue by niranjan94** [rancher/rancher#22053]
**What kind of request is this:** bug

**Steps to reproduce:**
- Wait until certificates are about to expire and rancher UI shows `This cluster has certs that are expiring or have expired.`
- Rotate certificates through the Rancher UI

**Result:**
The `This cluster has certs that are expiring or have expired.` message still remains on the UI even after certificate rotation.

<img width="1920" alt="Screenshot 2019-08-08 at 11 43 33 AM" src="https://user-images.githubusercontent.com/2404372/62679508-4bf30800-b9d2-11e9-9216-a07500175377.png">
 

**Other details that may be helpful:**
First noticed this issue when the expiry status was exposed in the rancher UI. 
Still exists after updating to v2.2.7

**Environment information**
- Rancher version: `v2.2.7`
- Installation option: Single install

**Cluster information**
- Cluster type: Infrastructure Provider
- Machine type: cloud
- Kubernetes version:

```
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.5", GitCommit:"0e9fcb426b100a2aea5ed5c25b3d8cfbb01a8acf", GitTreeState:"clean", BuildDate:"2019-08-05T09:13:08Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version:

```
Server:
 Version:      17.03.2-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 01:35:00 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by niranjan94** [comment]
Hi 😄 Any updates/workarounds for this ? (Thanks in advance 🎉)

**Comment by niranjan94** [comment]
Hi ... I can confirm that this issue existing on the latest v2.3. Any possible fixes ?

**Comment by loganhz** [comment]
@niranjan94 Could you provide the view API result for one of your cluster? 

It should be the same issue with https://github.com/rancher/rancher/issues/24333

**Comment by niranjan94** [comment]
@loganhz is this helpful ? Do let me know if any additional information is required.

<img width="343" alt="Screenshot 2019-12-19 at 10 50 08 AM" src="https://user-images.githubusercontent.com/2404372/71147048-8d7fa380-224d-11ea-8c6f-6d0f37d3fcfd.png">

**Comment by loganhz** [comment]
![image](https://user-images.githubusercontent.com/21168270/71158079-36de8d80-227e-11ea-90e1-d630d498bb1f.png)

This is the node which is not in your cluster any more. Correct?

**Comment by niranjan94** [comment]
> This is the node which is not in your cluster any more. Correct?

Yep. Correct. That node has been completely removed from the cluster.

> If so, the issue will be fixed in #24333

That is awesome ! Thanks for the update @loganhz :smile:

**Comment by loganhz** [comment]
Thanks @niranjan94 !
Close in favor of https://github.com/rancher/rancher/issues/24333
